src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -437,6 +438,19 @@ await Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions.Si
         {
             _app.MapPost("/authentication/logout", () => TypedResults.SignOut(authenticationSchemes: [CookieAuthenticationDefaults.AuthenticationScheme, OpenIdConnectDefaults.AuthenticationScheme]));
         }
+
+        _app.MapGet("/api/set-language", (string? language, string redirectUrl, HttpContext httpContext) =>
+        {
+            if (language is not null)
+            {
+                httpContext.Response.Cookies.Append(
+                    CookieRequestCultureProvider.DefaultCookieName,
+                    // keep current culture for formatting, etc., but change ui (display) culture
+                    CookieRequestCultureProvider.MakeCookieValue(new RequestCulture(CultureInfo.CurrentCulture.Name, language)));
```

Comment:
Right now this is a session cookie and is lost when the browser is complete closed. Add an expiry:

> new CookieOptions { Expires = DateTimeOffset.UtcNow.AddYears(1) }